smithy-kotlin-codegen/src/test/kotlin/software/amazon/smithy/kotlin/codegen/rendering/protocol/HttpProtocolClientGeneratorTest.kt

```diff
@@ -196,7 +196,7 @@ class HttpProtocolClientGeneratorTest {
         }
         op.install(MockMiddleware(configurationField1 = "testing"))
         val rootSpan = config.tracer.createRootSpan("GetFooStreamingInputNoOutput-${'$'}{op.context.sdkRequestId}")
-        return op.context.withRootSpan(rootSpan) {
+        return coroutineContext.withRootTraceSpan(rootSpan) {
```

Comment:
This seems like we don't need a wrapper necessarily for a root span, wouldn't this suffice:

```kotlin
    return withContext(TraceSpanContext(rootSpan)) { ... }
```

Reply:
The `withRootTraceSpan` method also verifies that no other trace span is active and then closes the root when the lambda is finished.